app/src/main/java/org/dhis2/usescases/main/MainActivity.kt

```diff
@@ -195,13 +195,17 @@ class MainActivity :
         }
 
         presenter.observeDataSync().observe(this) {
-            if (it.firstOrNull()?.state == WorkInfo.State.SUCCEEDED) {
+            if (it.firstOrNull()?.state == WorkInfo.State.RUNNING) {
+                setFilterButtonVisibility(false)
+                setBottomNavigationVisibility(false)
+            } else if (
+                it.firstOrNull()?.state == WorkInfo.State.SUCCEEDED ||
+                it.firstOrNull()?.state == WorkInfo.State.FAILED ||
+                it.firstOrNull()?.state == WorkInfo.State.CANCELLED
+            ) {
                 setFilterButtonVisibility(true)
                 setBottomNavigationVisibility(true)
                 presenter.onDataSuccess()
```

Comment:
@Balcan presenter.onDataSuccess() will be called and set to true even if FAILED

Reply:
True! I'm changing it :)